src/middlewared/middlewared/plugins/ldap.py

```diff
@@ -613,10 +644,21 @@ def get_dn(self, dn=None, ldap=None):
         """
         Returns False if disabled, True if healthy, raises exception if faulted.
         """
-        if not (await self.config())['enable']:
+        verrors = ValidationErrors()
+        ldap = await self.config()
+        if not ldap['enable']:
             return False
 
-        ldap = await self.config()
+        await self.common_validate(ldap, ldap, verrors)
+        if verrors:
+            await self.middleware.call(
+                'datastore.update',
+                'directoryservice.ldap',
+                ldap['id'],
+                {'ldap_enable': False}
+            )
+            cerrors = ' '.join(v.errmsg for v in verrors.errors)
+            raise CallError(cerrors, errno.EINVAL)
```

Comment:
And same here, perhaps just `raise verrors` would do the trick here? Not understanding why this should be a CallError?